src/main/java/com/kdt/localinfo/common/BaseEntity.java

```diff
@@ -0,0 +1,31 @@
+package com.kdt.localinfo.common;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import lombok.Getter;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.Column;
+import javax.persistence.EntityListeners;
+import javax.persistence.MappedSuperclass;
+import java.time.LocalDateTime;
+
+@Getter
+@EntityListeners(value = AuditingEntityListener.class)
+@MappedSuperclass
+public class BaseEntity {
+
+    @CreatedDate
+    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
+    @Column(name = "created_at", nullable = false)
+    protected LocalDateTime createdAt;
+
+    @LastModifiedDate
+    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
+    @Column(name = "updated_at", nullable = false)
+    protected LocalDateTime updatedAt;
+
+    @Column(name = "deleted_at")
+    protected LocalDateTime deletedAt;
```

Comment:
```suggestion
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @PrePersist
    void onPrePersist() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        this.createdAt = LocalDateTime.parse(LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")), formatter);
        this.updatedAt = createdAt;
    }

    @PreUpdate
    void onPreUpdate() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        this.updatedAt = LocalDateTime.parse(LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")), formatter);
    }
```

@choiyoungkwon12
찾아보니까 이 부분을 `@PrePersist`, `@PreUpdate`를 사용하면 될 것 같아요.
formatter 써서 String으로 형식 변경하고 다시 그 형식으로 LocalDateTime 타입으로 파싱하면 잘 바뀝니다!
엔티티 생성 전에는 updatedAt을 createdAt과 동일한 값을 넣어주고 수정이 일어나면 그때 바뀐 날짜로 넣으면 될 것 같아요. `@CreatedDate`, `@LastModifiedDate` 어노테이션들이 JsonFormat 안먹는다고 하더라고요!

Reply:
아 추가로 혹시 접근제한자를 protected로 한 특별한 이유가 있을까요?